test/jasmine/tests/mesh3d_test.js

```diff
@@ -3,55 +3,225 @@ var createGraphDiv = require('../assets/create_graph_div');
+
+            Plotly.plot(gd, [{
+                type: 'mesh3d',
+                x: [0, 1, 2, 0],
+                y: [0, 0, 1, 2],
+                z: [0, 2, 0, 1],
+                i: [0, 0, 0, 1],
+                j: [1, 2, 3, 2],
+                k: [2, 3, 1, 3],
+                intensity: [0, 0.33, 0.66, 3]
+            }])
+            .then(function() {
+                _assert([undefined, undefined, undefined], [true, 0, 3]);
+
+                return Plotly.restyle(gd, 'cmin', 0);
+            })
+            .then(function() {
+                _assert([false, 0, undefined], [false, 0, 3]);
+
+                return Plotly.restyle(gd, 'cmax', 10);
+            })
+            .then(function() {
+                _assert([false, 0, 10], [false, 0, 10]);
+
+                return Plotly.restyle(gd, 'cauto', true);
+            })
+            .then(function() {
+                _assert([true, 0, 10], [true, 0, 3]);
+
+                return Plotly.purge(gd);
+            })
+            .catch(failTest)
+            .then(done);
+        });
+    });
+
+    describe('dimension and expected visibility tests', function() {
+        var gd;
+
+        beforeEach(function() {
+            gd = createGraphDiv();
+        });
+
+        afterEach(function() {
+            Plotly.purge(gd);
+            destroyGraphDiv();
+        });
+
+        function assertVisibility(exp, msg) {
+            expect(gd._fullData[0]).not.toBe(undefined, 'no visibility!');
+            expect(gd._fullData[0].visible).toBe(exp, msg);
         }
 
-        Plotly.plot(gd, [{
-            type: 'mesh3d',
-            x: [0, 1, 2, 0],
-            y: [0, 0, 1, 2],
-            z: [0, 2, 0, 1],
-            i: [0, 0, 0, 1],
-            j: [1, 2, 3, 2],
-            k: [2, 3, 1, 3],
-            intensity: [0, 0.33, 0.66, 3]
-        }])
-        .then(function() {
-            _assert([undefined, undefined, undefined], [true, 0, 3]);
-
-            return Plotly.restyle(gd, 'cmin', 0);
-        })
-        .then(function() {
-            _assert([false, 0, undefined], [false, 0, 3]);
-
-            return Plotly.restyle(gd, 'cmax', 10);
-        })
-        .then(function() {
-            _assert([false, 0, 10], [false, 0, 10]);
-
-            return Plotly.restyle(gd, 'cauto', true);
-        })
-        .then(function() {
-            _assert([true, 0, 10], [true, 0, 3]);
-
-            return Plotly.purge(gd);
-        })
-        .catch(failTest)
-        .then(done);
+        it('@gl mesh3d should be visible when the indices are not integer', function(done) {
+            Plotly.plot(gd, [{
+                x: [0, 1, 0.5, 0.5],
+                y: [0, 0.5, 1, 0.5],
+                z: [0, 0.5, 0.5, 1],
+                i: [0, 0, 0, 1.00001],
+                j: [1, 1, 2, 1.99999],
+                k: [2, 3, 3, 3.00001],
+                type: 'mesh3d'
+            }])
+            .then(function() {
+                assertVisibility(true, 'to be visible');
```

Comment:
Right, this is correct: `fullData` will have `visible:false`. But this doesn't test any of the new logic in convert.js.

Could you add a check for e.g.

```
gd._fullLayout.scene._scene.glplot.objects[0].positions
```

which should be an ok proxy.

Reply:
OK. Will do so.

Reply:
Thanks, sorry for not catching this sooner.

Reply:
No problem. Previously we get different visibility values there...